hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountRecordsHandler.java

```diff
@@ -51,20 +63,46 @@ public QueryHeader extractHeader(@NonNull final Query query) {
     @Override
     public Response createEmptyResponse(@NonNull final ResponseHeader header) {
         requireNonNull(header);
-        final var response = CryptoGetAccountRecordsResponse.newBuilder().header(requireNonNull(header));
+        final var response = CryptoGetAccountRecordsResponse.newBuilder().header(header);
         return Response.newBuilder().cryptoGetAccountRecords(response).build();
     }
 
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final var query = context.query();
+        final var op = query.cryptoGetAccountRecordsOrThrow();
```

Comment:
Should only use orThrow if we have already verified we have the item expected.
if `query.hasCryptoGetAccountRecords()` is false then this should throw a `PreCheckException`, not `NullPointerException` (as the `orThrow` method does).

Reply:
This seems to defeat the point of `orThrow` to me, since we'll get NPE if we ever try to call a method on `op` anyway. I guess I need to clarify our throwing strategy.

Reply:
Changed to a precheck exception